components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -1081,14 +1082,45 @@ public int compare(Tag o1, Tag o2) {
 		                                APIConstants.TAGS_INFO_ROOT_LOCATION +
 		                                        "/%s/description.txt";
 		String thumbnailPathPattern = APIConstants.TAGS_INFO_ROOT_LOCATION + "/%s/thumbnail.png";
+
+		//if the tenantDomain is not specified super tenant domain is used
+		if(tenantDomain == null || tenantDomain.trim() == "" ){
+			try {
+				tenantDomain = ServiceReferenceHolder.getInstance().getRealmService().getTenantManager().getSuperTenantDomain();
+			} catch (org.wso2.carbon.user.core.UserStoreException e) {
+				handleException("Cannot get super tenant domain name",e);
+			}
+		}
+
+		//get the registry instance related to the tenant domain
+		UserRegistry govRegistry = null;
+		try {
+			ServiceReferenceHolder.getInstance().getRealmService().getTenantManager().getTenantId(tenantDomain);
+			int tenantId = ServiceReferenceHolder.getInstance().getRealmService().getTenantManager()
+			                                     .getTenantId(tenantDomain);
+			RegistryService registryService =
+					ServiceReferenceHolder.getInstance()
+					                      .getRegistryService();
+			govRegistry = registryService.getGovernanceSystemRegistry(tenantId);
+
+		} catch (UserStoreException e) {
+			handleException("Cannot get tenant id for tenant domain name:"+tenantDomain,e);
+		} catch (RegistryException e) {
+			handleException("Cannot get registry for tenant domain name:"+tenantDomain,e);
+		}
+
+		if (govRegistry != null) {
 		for (Tag tag : tags) {
 			// Get the description.
 			Resource descriptionResource = null;
 			String descriptionPath = String.format(descriptionPathPattern, tag.getName());
 			try {
-				descriptionResource = registry.get(descriptionPath);
+				if (govRegistry.resourceExists(descriptionPath)) {
+					descriptionResource = govRegistry.get(descriptionPath);
+				}
 			} catch (RegistryException e) {
-				log.warn(String.format("Cannot get the description for the tag '%s'", tag.getName()));
+				//warn and proceed to the next tag
+				log.warn(String.format("Error while querying the existence of the description for the tag '%s'", tag.getName()));
```

Comment:
Fixed in #1494(https://github.com/wso2/carbon-apimgt/commit/b672e7ec2a5973ae34af2999490837d9333cafc4)